packages/@glimmer/runtime/lib/vm/arguments.ts

```diff
@@ -544,21 +454,72 @@ class CapturedBlockArgumentsImpl implements CapturedBlockArguments {
   }
 }
 
-export class CapturedArgumentsImpl implements CapturedArguments {
-  constructor(
-    public positional: CapturedPositionalArguments,
-    public named: CapturedNamedArguments,
-    public length: number
-  ) {}
+export function capturedArgs(named: Dict<PathReference>, positional: PathReference[]) {
```

Comment:
The name here seems a tad odd to me, shouldn't it be `captureArgs`?

Reply:
maybe `createCapturedArgs`? It's not exactly capturing them, it's making them manually

Reply:
I see, ya the thing that seemed odd was the past tense phrasing. `createCapturedArgs` seems good to me.